force-app/main/default/classes/StaticTask.cls

```diff
@@ -1,38 +1,106 @@
 public with sharing class StaticTask {
-  /*This method recieves a string payload with contact data that loks like  
+  /*This method recieves a string payload with contact data that looks like  
 
   'FirstName: firstname, LastName: lastname, Email: ex@mple.com;
    FirstName: firstname, LastName: lastname, Email: ex@mple.com;
    ...
   '
 
-  . Then it parses data and return Map collection where key is meail and value is Contact
-  Feel free to refactorthis class. Don't forget about single responsibility approach when you create a new method.
+  . Then it parses data and return Map collection where key is e-mail and value is Contact
+  Feel free to refactor this class. Don't forget about single responsibility approach when you create a new method.
   */
-  public static Map<String, Contact> extractEmailToContactMap(String payload) {
-    Map<String, Contact> emailToContactMap = new Map<String, Contact>();
-    List<String> contacts = payload.split(';');
-    for (String contactAsString : contacts) {
-      Contact parsedContact = new Contact();
-      List<String> contactSeparatedByComaString = contactAsString.split(',');
-      for (String element : contactSeparatedByComaString) {
-        if (element.contains('FirstName')) {
-          parsedContact.FirstName = element.split(':')[1].replaceAll(' ', '');
-        }
-        if (element.contains('LastName')) {
-          parsedContact.LastName = element.split(':')[1].replaceAll(' ', '');
-        }
-        if (element.contains('Email')) {
-          if (String.isNotBlank(element.split(':')[1])) {
-            parsedContact.Email = element.split(':')[1].replaceAll(' ', '');
-            emailToContactMap.put(
-              element.split(':')[1].replaceAll(' ', ''),
-              parsedContact
-            );
-          }
-        }
+
+  public static Map<String, Contact> extractEmailToContactMap (String payload) {
+
+    List<String> payloadAsStringList = StaticTask.splitPayloadByElementStringList(payload);
+    Map<Integer, List<String>> sObjectFieldsMap = StaticTask.splitStringListByObjectFieldMap(payloadAsStringList);
+    List<Contact> contactList = StaticTask.splitObjectFieldMapByRecords(sObjectFieldsMap);
```

Comment:
this is nice. Might not be optimal, since you have to go over the same list twice (O(2n) vs O(n)), but it's very easy to read.

Sometimes refactoring (making code readable) and optimization (making code run faster) don't get along. In my opinion, clean code is more important in this cases, since you wright code for other developers to read.